src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingService.cs

```diff
@@ -73,13 +79,104 @@ internal static LanguageService LanguageServiceInstance
         /// </summary>
         /// <param name="serviceHost"></param>
         /// <param name="context"></param>
         public void InitializeService(ServiceHost serviceHost)
         {
             serviceHost.SetRequestHandler(ScriptingScriptAsRequest.Type, HandleScriptingScriptAsRequest);
+            serviceHost.SetRequestHandler(ScriptingRequest.Type, this.HandleScriptExecuteRequest);
+            serviceHost.SetRequestHandler(ScriptingCancelRequest.Type, this.HandleScriptCancelRequest);
+            serviceHost.SetRequestHandler(ScriptingListObjectsRequest.Type, this.HandleListObjectsRequest);
+
+            // Register handler for shutdown event
+            serviceHost.RegisterShutdownTask((shutdownParams, requestContext) =>
+            {
+                this.Dispose();
+                return Task.FromResult(0);
+            });
+        }
+
+        /// <summary>
+        /// Handles request to execute start the list objects operation.
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Await.Warning", "CS4014:Await.Warning", Justification = "Not using await for ScriptingOperation.Execute() since this a long running operation.")]
+        private async Task HandleListObjectsRequest(ScriptingListObjectsParams parameters, RequestContext<ScriptingListObjectsResult> requestContext)
+        {
+            try
+            {
+                ScriptingOperation operation = new ScriptingListObjectsOperation(parameters, requestContext);
+                this.ActiveOperations[operation.OperationId] = operation;
+                await requestContext.SendResult(new ScriptingListObjectsResult { OperationId = operation.OperationId });
+#pragma warning disable 4014
+                operation.Execute().ContinueWith((t) => this.ActiveOperations.TryRemove(operation.OperationId, out operation));
+#pragma warning restore 4014
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e);
+            }
+        }
+
+        /// <summary>
+        /// Handles request to execute start the script operation.
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Await.Warning", "CS4014:Await.Warning", Justification = "Not using await for ScriptingOperation.Execute() since this a long running operation.")]
+        public async Task HandleScriptExecuteRequest(ScriptingParams parameters, RequestContext<ScriptingResult> requestContext)
+        {
+            try
+            {
+                ScriptingOperation operation = new ScriptingScriptOperation(parameters, requestContext);
+                this.ActiveOperations[operation.OperationId] = operation;
+                await requestContext.SendResult(new ScriptingResult { OperationId = operation.OperationId });
+#pragma warning disable 4014
+                operation.Execute().ContinueWith((t) => this.ActiveOperations.TryRemove(operation.OperationId, out operation));
```

Comment:
Per offline discussion, our best practice is to a) put this in its own method similar to ConnectionService.RunConnectRequestHandlerTask, and b) kick off this request before doing SendResult (e.g. move this line above `equestContext.SendResult`. Code would look similar to
```
Task.Run(async () => 
            {
                try
                {
                    await operation.Execute();
                    this.ActiveOperations.TryRemove(operation.OperationId, out operation);
                }
                catch (Exception ex)
                {
                    // either send event with failure status, or call SendError if these fail
                    await ServiceHost.SendEvent(ConnectionCompleteNotification.Type, result);
                }
            });
```

Thanks!